contracts/CourtRoom.sol

```diff
@@ -7,24 +7,32 @@ import "./ensabstract.sol";
 
 contract CaseContract is Owned {
 
+  struct promise{
+		address beneficiary;
+		uint256 blockNumber;
+		uint8 sig_v;
+		bytes32 sig_r;
+		bytes32 sig_s;
+	}
 	struct claim {
 		bytes32 claimId;
 		address plaintiff;
 		bytes32[] evidence;
+		promise   servicePromise;
 		uint status;
 		bool valid;
 	}
 
 	//claimid map to claim
-  mapping(bytes32 => claim) public OpenClaims;
+  mapping(bytes32 => claim)  OpenClaims;
 
 
 	function CaseContract() {
 	}
 
 	function newClaim(address _plaintiff, bytes32 _evidence) returns (bytes32 claimId) {
 
-	  claim clm;
+	  claim storage clm;
```

Comment:
Why is the clm being put into storage? Surely memory is more fitting as the clm is then stored in the OpenClaims mapping, which is in storage?